bentoml/configuration/__init__.py

```diff
@@ -61,6 +61,7 @@ def parameterized_config(template):
 
 DEFAULT_BENTOML_HOME = expand_env_var(os.environ.get("BENTOML_HOME", "~/bentoml"))
 BENTOML_HOME = DEFAULT_BENTOML_HOME
+BENTOML_CONFIG = os.path.join(BENTOML_HOME, "bentoml.yml")
```

Comment:
This is interesting, it means users will be able to put a default config file under the BENTOML_HOME directory which is expected to affect all BentoML usage on that machine?

For configs that affect the BentoML usage during the model packaging stage, does a regular `import bentoml` pick up this default config file? And how do users set a config file in that scenario where no CLI command is involved?

We talked about adding a python API for setting a config before, I think it could be something like this:

```python
import bentoml
bentoml.set_config("~/foo/bar/my_bentoml_config.yml")

@env(...)
@artifacts(...)
class MyClassifier(bentoml.BentoService):

     @api(...)
     def classify(self, ...):
             ...
```
And in the `bentoml/__init__.py` file, we can have it load the default config in BENTOML_HOME, with `bentoml.set_config(benotml.configuration.BENTOML_CONFIG)`.

Reply:
> This is interesting, it means users will be able to put a default config file under the BENTOML_HOME directory which is expected to affect all BentoML usage on that machine?

The goal behind reading config from BentoML home is that, when user containerize with a config option, the config will be placed under BentoML home and be automatically picked up when the docker image is run.
```zsh
$ bentoml containerize bento --config ~/foo/bar/bentoml.yml
```

> For configs that affect the BentoML usage during the model packaging stage, does a regular import bentoml pick up this default config file? And how do users set a config file in that scenario where no CLI command is involved?

We have the option to include model packaging stage in the scope of the config. Do you see a need to have config there as the packaging logic is pretty standard and non-customizable.

> We talked about adding a python API for setting a config before, I think it could be something like this:

To package a config into a bundle, I was thinking something like the following.

```py
import bentoml
from iris_classifier import IrisClassifier  # example service from the quick start guide

model = ... # training a model

svc = IrisClassifier()
svc.pack(name='model', model=model, config="~/foo/bar/my_bentoml_config.yml")
svc.save()
```